nomad/service_registration_endpoint.go

```diff
@@ -381,12 +339,12 @@ func (s *ServiceRegistration) GetService(
 	}
 	defer metrics.MeasureSince([]string{"nomad", "service_registration", "get_service"}, time.Now())
 
-	aclObj, err := s.srv.ResolveClientOrACL(args)
+	aclObj, err := s.srv.ResolveACL(args)
 	if err != nil {
 		return structs.ErrPermissionDenied
 	}
-	if args.GetIdentity().Claims == nil &&
-		!aclObj.AllowNsOp(args.RequestNamespace(), acl.NamespaceCapabilityReadJob) {
+	if aclObj != nil && !aclObj.AllowServiceRegistrationReadList(args.RequestNamespace(),
+		args.GetIdentity().Claims != nil) {
 		return structs.ErrPermissionDenied
 	}
```

Comment:
didn't we want to avoid allowing `aclObj == nil` (or is that coming later)?

Reply:
The `ACLsDisabled` object is coming later, but you're right we don't need this here now because `AllowServiceRegistrationReadList` handles that correctly (with a TODO to reverse the test for a nil ACL once we do have the `ACLsDisabled`. Will fix.

Reply:
Done in 9701aca